pkg/ignore/types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ignore
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Ignorer  helps identifying if a BundleFile needs to be ignored
+type Ignorer interface {
+	// Ignore checks file has to be ignored or not, returns true if it needs to be ignored
+	Ignore(path string, fi os.FileInfo) Ignoreable
+	// Patterns returns the pattern associated with interface
+	Patterns() *sets.String
```

Comment:
I don't think we use this anymore?

Reply:
yeah, it slipped through my refactor, will include it as part of next push :)